aws/data_source_aws_guardduty_detector.go

```diff
@@ -0,0 +1,95 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsGuarddutyDetector() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsGuarddutyDetectorRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"service_role_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"finding_publishing_frequency": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsGuarddutyDetectorRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	detectorId := d.Get("id").(string)
+
+	if detectorId == "" {
+		log.Print("[DEBUG] No Guardduty ID has been defined, try to lookup using find (There should be only one)")
+		input := &guardduty.ListDetectorsInput{}
+
+		resp, err := conn.ListDetectors(input)
+		if err != nil {
+			return err
+		}
+
+		if resp == nil || len(resp.DetectorIds) == 0 {
+			return fmt.Errorf("no detectors found")
+		}
+		if len(resp.DetectorIds) > 1 {
+			return fmt.Errorf("multiple detectors found; Amazon AWS API behavior has changed; this is a bug in the provider an should be reported")
```

Comment:
If the API behavior does change to allow multiple GuardDuty Detectors in a region, what would we change in this data source? I think we should instead encourage using the `id` argument and allow any feature requests for a plural `aws_guardduty_detectors` data source to come in naturally, if necessary. 👍 

```suggestion
			return fmt.Errorf("multiple GuardDuty Detectors found; please use the `id` argument to look up a single detector")
```

Reply:
The error message has been updated.